components/automate-ui/src/app/page-components/insight-attributes-dropdown/insight-attributes-dropdown.component.ts

```diff
@@ -0,0 +1,113 @@
+import { isEqual } from 'lodash/fp';
+
+// Animations
+const menuOpenAnim = trigger('menuOpenAnim', [
+  transition(':enter', [
+    style({ opacity: 0, transform: 'translateY(-8px) scale(0)' }),
+    group([
+      query('@childAnim', animateChild()),
+      animate('220ms ease-out', style({ opacity: 1, transform: 'translateY(0px) scale(1)' }))
+    ])
+  ]),
+  transition(':leave', [
+    style({ opacity: 1 }),
+    animate('195ms ease-in', style({ opacity: 0 }))
+  ])
+]);
+
+const childAnim = trigger('childAnim', [
+  transition(':enter', [
+    style({ opacity: 0 }),
+    animate('50ms 170ms', style({ opacity: 1 }))
+  ])
+]);
+
+@Component({
+  selector: 'app-insight-attributes-dropdown',
+  templateUrl: './insight-attributes-dropdown.component.html',
+  styleUrls: ['./insight-attributes-dropdown.component.scss'],
+  animations: [ menuOpenAnim, childAnim ]
+})
+export class InsightAttributesDropdownComponent implements OnInit {
+
+  @Input() saveAsDefault = true;
+  @Input() lastSelectedOptions: string[] = []; // these are filter ids
+  @Output() onUpdateFilters: EventEmitter<any> = new EventEmitter();
+  @Output() onToggleMenu: EventEmitter<any> = new EventEmitter();
+
+  @HostBinding('@menuOpenAnim') menuOpenAnim;
+
+  public options: FilterOption[] = FilterableOptions;
+  public selectedOptions: string[] = []; // these are filter ids
+  private initialDefaultSavePref = false;
+  public hasNewValues = false;
+  public saveDefaultChanged = false;
+
+  ngOnInit() {
+    // creating a reference to fall back on
+    // will likely change after we're storing these away elsewhere
+    this.lastSelectedOptions = [...this.selectedOptions];
+    this.initialDefaultSavePref = this.saveAsDefault;
+  }
+
+  get disableSubmit() {
+    return this.saveDefaultChanged || this.hasNewValues ? false : true;
+  }
+
+  public handleSelect(event: Event) {
+    const target = event.target as HTMLElement;
+    const filter = target.getAttribute('data-filterValue');
+
+    const isFilteredIndex = this.selectedOptions.indexOf(filter);
+    if ( isFilteredIndex >= 0 ) {
+      this.selectedOptions.splice(isFilteredIndex, 1);
+      this.updateButtonState(target, 'off');
+    } else {
+      if (this.selectedOptions.length < 5) {
+        this.selectedOptions.push(filter);
+        this.updateButtonState(target, 'on');
+      }
+    }
+
+    // check if form should be valid based on new selected options
+    this.hasNewValues = !isEqual(this.lastSelectedOptions.sort(), this.selectedOptions.sort());
+  }
+
+  public handleUpdate(): void {
+    // emit new selected filter ids
+    this.onUpdateFilters.emit({
+      filters: this.selectedOptions,
+      saveAsDefault: this.saveAsDefault
+    });
+    this.onToggleMenu.emit();
+  }
+
+  public handleCancel(): void {
+    this.selectedOptions.forEach(option => {
+      const target = document.querySelector('.filter-button[data-filterValue="' + option + '"]');
```

Comment:
Prefer using string builder instead of concatenation:
```suggestion
      const target = document.querySelector(`.filter-button[data-filterValue="${option}"]`);
```